platform/posix/transport/include/plaintext_posix.h

```diff
@@ -0,0 +1,123 @@
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef PLAINTEXT_POSIX_H_
+#define PLAINTEXT_POSIX_H_
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Logging related header files are required to be included in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define LIBRARY_LOG_NAME and  LIBRARY_LOG_LEVEL.
+ * 3. Include the header file "logging_stack.h".
+ */
+
+/* Include header that defines log levels. */
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME     "TCP"
+#endif
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_DEBUG
+#endif
+
+#include "logging_stack.h"
+
+/************ End of logging configuration ****************/
+
+/* Transport includes. */
+#include "transport_interface.h"
+#include "sockets_posix.h"
+
+/**
+ * @brief Definition of the network context.
+ *
+ * @note An integer is used to store the descriptor of the socket.
+ */
+struct NetworkContext
+{
+    int socketDescriptor;
+};
+
+/**
+ * @brief Establish TCP connection to server.
+ *
+ * @param[in] pNetworkContext The network context.
+ * @param[in] pServerInfo Server connection info.
+ * @param[in] sendTimeout Timeout for transport send.
+ * @param[in] recvTimeout Timeout for transport recv.
+ *
+ * @note A timeout of 0 means infinite timeout.
+ *
+ * @return #SOCKETS_SUCCESS if successful;
+ * #SOCKETS_INVALID_PARAMETER, #SOCKETS_DNS_FAILURE, #SOCKETS_CONNECT_FAILURE on error.
+ */
+SocketStatus_t Plaintext_Connect( NetworkContext_t pNetworkContext,
+                                  const ServerInfo_t * pServerInfo,
+                                  uint32_t sendTimeoutMs,
+                                  uint32_t recvTimeoutMs );
+
+/**
+ * @brief Close TCP connection to server.
+ *
+ * @param[in] pNetworkContext The network context.
+ *
+ * @return #SOCKETS_SUCCESS if successful; #SOCKETS_INVALID_PARAMETER on error.
+ */
+SocketStatus_t Plaintext_Disconnect( const NetworkContext_t pNetworkContext );
+
+/**
+ * @brief The transport receive function that defines the transport interface.
+ *
+ * This is passed as the #TransportInterface.recv function used for reading
+ * data received from the network.
+ *
+ * @param[in] pNetworkContext User defined context (TCP socket descriptor).
+ * @param[out] pBuffer Buffer to read network data into.
+ * @param[in] bytesToRead Number of bytes requested from the network.
+ *
+ * @return Number of bytes received if successful; negative value on error.
+ */
+int32_t Plaintext_Recv( NetworkContext_t pNetworkContext,
+                        void * pBuffer,
+                        size_t bytesToRecv );
+
+/**
```

Comment:
```
/**
 * @brief Sends data over an established TCP connection.
 *
 * This can be used as the #TransportInterface.send function to send data over
 * the network.
```